test/e2e/upgrades/apps/deployments.go

```diff
@@ -111,60 +108,63 @@ func (t *DeploymentUpgradeTest) Setup(f *framework.Framework) {
 
 	rs, err = deploymentutil.GetNewReplicaSet(deployment, c.ExtensionsV1beta1())
 	framework.ExpectNoError(err)
-	if rs == nil {
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set for deployment %q", deployment.Name))
-	}
-
-	if rs.UID == t.oldRS.UID {
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set different from the previous one"))
+	if rs.UID == oldRSUID {
+		framework.ExpectNoError(fmt.Errorf("expected a new replicaset different from the previous one"))
 	}
 
-	// Store new replica set - should be the same after the upgrade.
-	t.newRS = rs
-	deployment, err = c.Extensions().Deployments(ns).Get(deployment.Name, metav1.GetOptions{})
+	t.newRSUID = rs.UID
+	deployment, err = deploymentClient.Get(deployment.Name, metav1.GetOptions{})
 	framework.ExpectNoError(err)
 	t.oldD = deployment
 }
 
-// Test checks whether the replica sets for a deployment are the same after an upgrade.
+// Test checks whether the replicasets for a deployment are the same after an upgrade.
 func (t *DeploymentUpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
 	// Block until upgrade is done
-	By(fmt.Sprintf("Waiting for upgrade to finish before checking replica sets for deployment %q", t.oldD.Name))
+	By(fmt.Sprintf("Waiting for upgrade to finish before checking replicasets for deployment %q", t.oldD.Name))
 	<-done
 
 	c := f.ClientSet
+	name := t.oldD.Name
+	deploymentClient := c.ExtensionsV1beta1().Deployments(t.oldD.Namespace)
 
-	deployment, err := c.Extensions().Deployments(t.oldD.Namespace).Get(t.oldD.Name, metav1.GetOptions{})
+	deployment, err := deploymentClient.Get(name, metav1.GetOptions{})
 	framework.ExpectNoError(err)
 	t.updatedD = deployment
 
+	By(fmt.Sprintf("Checking UID to verify deployment %q survives upgrade", name))
+	if t.updatedD.UID != t.oldD.UID {
+		framework.ExpectNoError(fmt.Errorf("expected same deployment UID: %v got: %v", t.oldD.UID, t.updatedD.UID))
+	}
+
+	By(fmt.Sprintf("Verifying revision of the deployment %q is still 2", name))
+	if deployment.Annotations == nil || deployment.Annotations[deploymentutil.RevisionAnnotation] != "2" {
```

Comment:
nit: `deployment.Annotations[deploymentutil.RevisionAnnotation] != "2"` is enough.